src/CamScripterAPI.ts

```diff
@@ -0,0 +1,222 @@
+import * as WebSocket from 'ws';
+import * as EventEmitter from 'events';
+
+import { Digest } from './Digest';
+
+export type CamScripterOptions = {
+    tls?: boolean;
+    tlsInsecure?: boolean; // Ignore HTTPS certificate validation (insecure)
+    ip?: string;
+    port?: number;
+    auth?: string;
+};
+
+export type Declaration = {
+    type?: '' | 'SOURCE' | 'DATA';
+    namespace: string;
+    key: string;
+    value: string | boolean | number;
+    value_type: 'STRING' | 'INT' | 'BOOL' | 'DOUBLE';
+    key_nice_name?: string;
+    value_nice_name?: string;
+};
+
+export type EventDeclaration = {
+    declaration_id: string;
+    stateless: boolean;
+    declaration: Declaration[];
+};
+
+export type EventUndeclaration = {
+    declaration_id: string;
+};
+
+export type EventData = {
+    namespace: string;
+    key: string;
+    value: string | boolean | number;
+    value_type: 'STRING' | 'INT' | 'BOOL' | 'DOUBLE';
+};
+
+export type Event = {
+    declaration_id: string;
+    event_data: EventData[];
+};
+
+export type Response = {
+    call_id: number;
+    message: string;
+};
+
+type Message = {
+    call_id: number;
+    command: string;
+    data: any;
+};
+
+type AsyncMessage = {
+    resolve;
+    reject;
+};
+
+export class CamScripterAPI extends EventEmitter {
+    private tls: boolean;
+    private tlsInsecure: boolean;
+    private ip: string;
+    private port: number;
+    private auth: string;
+    private callId: number;
+    private sendMessages: Record<number, AsyncMessage>;
+
+    private ws: WebSocket = null;
+
+    constructor(options?: CamScripterOptions) {
+        super();
+
+        this.tls = options?.tls ?? false;
+        this.tlsInsecure = options?.tlsInsecure ?? false;
+        this.ip = options?.ip ?? '127.0.0.1';
+        this.port = options?.port ?? (this.tls ? 443 : 80);
+        this.auth = options?.auth ?? '';
+
+        this.callId = 0;
+        this.sendMessages = {};
+
+        EventEmitter.call(this);
+    }
+
+    async cameraEventsConnect() {
```

Comment:
I suggest to name this mechanism like `cameraEventsGenerator` (see the class name suggestion). The name of this function would be just plain `connect()`.

Reply:
fixed